src/API/Transaction/DetailedTransaction.php

```diff
@@ -21,12 +21,15 @@ public function __construct(array $data)
         $this->ip = isset($data['ip']) ? $data['ip'] : null;
         $this->ipCountry = isset($data['ip_country']) ? $data['ip_country'] : null;
 
+        $this->hasCardholderData = isset($data['has_cardholder_data']) ? $data['has_cardholder_data'] : false;
+        $this->cardholderData = isset($data['cardholder_data']) && null !== $data['cardholder_data'] ? new CardholderData($data['cardholder_data']) : null;
```

Comment:
This is utterly impossible to read, it has to be split out into multiple lines with normal if statements or the like.